### ISSUE TITLE
"Waiting for Peer" reasons

### ISSUE THREAD

**Issue by kaosbeat** [jacktrip/jacktrip#621]
Hi,

sorry for crossposting from the mailinglist over at https://groups.google.com/g/jacktrip-users/c/MguG7AqeTSE but I'm starting to feel this is more a dev than a users issue, or at least one that needs dev-understanding...

We're building a setup with a mixed OS audience (Raspbian/OSX). All run jacktrip 1.5.3

- we're running jacktrip -S on a RPi4.
- playout (speakers) are connected to this instance using jacktrip -C also on RPi4s.

This all works great

However, the person sending audio to the speaker connects to it using a Macbook (M1), is experiencing some difficulties.

We need to send 20 channels of audio from the macbook to 1 RPi with one seperate channel each
Like so:

```
                                    ┌───────────┐
                                    │RPi4 CM4 #1│
                                 ┌──┤jacktrip -C│--<<< speaker #1
                                 │  └───────────┘
                                 │
  ┌───────────┐    ┌───────────┐ │  ┌───────────┐
  │ MacBook M1├────┤RPi4 CM4   │ │  │RPi4 CM4 #2│
  │jacktrip -C│    │jacktrip -S├─┼──┤jacktrip -C│--<<< speaker #2
  │ 20Ch      │    └───────────┘ │  └───────────┘
  └───────────┘                 ...       ...           ...
                                 │  ┌───────────┐
                                 │  │RPi4  #20  │
                                 └──┤jacktrip -C│--<<< speaker #20
                                    └───────────┘
```

Because we don't care that much about latency, and more about reliability (oneway audio flow) Jack is configured on the speakers and the HUBserver as:

`jack sample rate 48000, frames/period 1024, period/buffer 16`

We're very happy with the audio quality...
however, the mac doens't like it.

When setting above jack settings, we can only send 4 channels to jacktrip, or it will just say Waiting for peer
![image](https://user-images.githubusercontent.com/204628/173342257-32f16d33-2fd3-4c33-bfe5-eb578f67dd04.png)

We tried 20 channels and it just keeps saying waiting for peer, not until we lower to 4 it will connect.

However if we lower the jack frames/period
![image](https://user-images.githubusercontent.com/204628/173342304-bd7d87ac-09e7-4274-90e6-b278fbecbe8b.png)

It will connect up to 17 channels (at 256 frames/period)
and if we even lower the frames/period further we can finally connect all 20 channels

Why is this?
Is this documented somewhere?

what is the best strategy to allow latency when sending a LOT of channels..

what are the limits for jacktrip in this regard. Should setup more Jacjtrip HUB servers? can I do this on a single RPi? Does QMaxThreads have something to do with this?

Basically I need to know if this "waiting for peer" can be solved without lowering frames/period. We want it this high to ensure no dropouts...

we already use -q 32 and -r 4

Thanks

**Comment by kaosbeat** [comment]
ok, sounds like an explanation.
I will do some experiments and report back
